pkg/kubelet/kuberuntime/kuberuntime_container.go

```diff
@@ -673,6 +672,10 @@ func (m *kubeGenericRuntimeManager) executePreStopHook(ctx context.Context, pod
 	case <-done:
 		klog.V(3).InfoS("PreStop hook completed", "pod", klog.KObj(pod), "podUID", pod.UID,
 			"containerName", containerSpec.Name, "containerID", containerID.String())
+	case <-ctx.Done():
+		klog.V(2).InfoS("Context expired while executing PreStop hook", "pod", klog.KObj(pod), "podUID", pod.UID,
+			"containerName", containerSpec.Name, "containerID", containerID.String(), "gracePeriod", gracePeriod)
+		return int64(metav1.Now().Sub(start.Time).Seconds())
```

Comment:
doesn't look to me like we need this? we'll fall through to the return directly below, which looks identical

Reply:
> doesn't look to me like we need this? we'll fall through to the return directly below, which looks identical

The logic here is the same as shortening the grace period.
If it's done.close that means it's not blocking normal execution
If it is ctx.Done(), it means to shorten the grace period and cancel the current blocking. ctx is passed from the previous layer kubelet.go in the previous layer pod_workers.go.

Reply:
I mean specifically the `return int64(metav1.Now().Sub(start.Time).Seconds())` seems duplicated with line 681 which directly proceeds this it seems

Reply:
> I mean specifically the `return int64(metav1.Now().Sub(start.Time).Seconds())` seems duplicated with line 681 which directly proceeds this it seems

Oh, yeah, you're right. I've made the changes.